tfjs-backend-webgpu/src/kernels/Concat_impl.ts

```diff
@@ -94,6 +94,20 @@ export function concatImpl(
     return outInfo;
   }
 
+  if (inputs.length >
+      env().getNumber('WEBGPU_MAX_STORAGE_BUFFER_IN_SHADER') - 1) {
+    const midIndex = Math.floor(inputs.length / 2);
+    const leftSide = concatImpl(inputs.slice(0, midIndex), axis, backend);
+    const rightSide = concatImpl(inputs.slice(midIndex), axis, backend);
+
+    const result = concatImpl([leftSide, rightSide], axis, backend);
+
+    backend.disposeData(leftSide.dataId);
+    backend.disposeData(rightSide.dataId);
+
+    return result;
+  }
+
```

Comment:
Maybe a more optimal algorithm is to group the inputs by (MAX_STORAGE_BUFFER_IN_SHADER - 1) recursively.

Reply:
Thanks for your help!
Previous code has a Array.push() -> Array.concat() mistake and loop number is wrong.
I followed Yang's suggestion and 800 inputs performance gained from 101.7ms -> 79.3ms
```
old:
LOG: 'Mean time: 117.462 ms -> 117.462 / rep'
Chrome 105.0.0.0 (Windows 10) LOG: 'Mean time: 117.462 ms -> 117.462 / rep'
LOG: 'Min time: 101.700 ms -> 101.700 / rep'
Chrome 105.0.0.0 (Windows 10) LOG: 'Min time: 101.700 ms -> 101.700 / rep'
```

```
new:
LOG: 'Mean time: 85.360 ms -> 85.360 / rep'
Chrome 105.0.0.0 (Windows 10) LOG: 'Mean time: 85.360 ms -> 85.360 / rep'
LOG: 'Min time: 79.300 ms -> 79.300 / rep'
Chrome 105.0.0.0 (Windows 10) LOG: 'Min time: 79.300 ms -> 79.300 / rep'
```
PTAL